### ISSUE TITLE
Multiple constraints on the same domain is reported as unparseable 

### ISSUE THREAD

**Issue by ethanrgrant** [sqlfluff/sqlfluff#3459]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

While linting the following statement:

```
CREATE DOMAIN mystr AS text
CONSTRAINT not_empty CHECK (LENGTH(value) > 0)
CONSTRAINT too_big CHECK (LENGTH(value) <= 50000);
```

I am getting an error that this is unparseable when i run :

`docker run -it --rm -v $PWD:/sql sqlfluff/sqlfluff:0.13.2 parse modeler/model.sql --dialect postgres`

```
L:  33 | P:  46 |  PRS | Line 33, Position 46: Found unparsable section: ',\n\tCONSTRAINT
                       | too_big CHECK (LENGTH(VALU...'
```

However, this is valid sql and should be parseable. This will work if i break it into statements

``` 
CREATE DOMAIN mystr
AS text CONSTRAINT not_empty CHECK (LENGTH(value) > 0);
ALTER DOMAIN mystr ADD CONSTRAINT too_big CHECK (LENGTH(value) <= 50000);
```

leading me to believe that it is failing because it does not expect multiple constraints to be possible

### Expected Behaviour

I would expect this to be parsed and properly understand that there are multiple constraints on a single domain.

### Observed Behaviour

A spurious unparseable error appeared

### How to reproduce

Can reproduce by running the following in the online sqlfuff playground with postgres as the dialect

```
CREATE DOMAIN mystr AS text
CONSTRAINT not_empty CHECK (LENGTH(value) > 0)
CONSTRAINT too_big CHECK (LENGTH(value) <= 50000);
```

### Dialect

postgres

### Version

0.13.2

### Configuration

used default configuration

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by tunetheweb** [comment]
Fix proposed in #3460
